home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/repository/ApartmentInvitationRepository.java

```diff
@@ -2,13 +2,18 @@
 
 import java.util.List;
 
-import com.softserveinc.ita.homeproject.homedata.entity.Apartment;
 import com.softserveinc.ita.homeproject.homedata.entity.ApartmentInvitation;
 import com.softserveinc.ita.homeproject.homedata.entity.InvitationStatus;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.data.repository.PagingAndSortingRepository;
 
+
+
 public interface ApartmentInvitationRepository extends PagingAndSortingRepository<ApartmentInvitation, Long>,
-        JpaSpecificationExecutor<Apartment> {
+        JpaSpecificationExecutor<ApartmentInvitation> {
+
+    List<ApartmentInvitation> findAllBySentDatetimeIsNullAndApartmentNotNullAndStatusEquals(InvitationStatus status);
```

Comment:
Oh no, Are any other possibility to query it? The method name looks ugly. What do you think about using QueryDsl or just manually form the spec use it here?

Reply:
Somewhere I heard that in Java, the more detailed the name, the better))
I will redo

Reply:
Yes, it is. The name of this approach is self-documented code. But dynamic DSL itself has several architectural gaps. And the major one I think is that the open-closed principle is broken together with the impossibility to follow the interface-segregation principle.

I suggest taking a look at https://www.baeldung.com/intro-to-querydsl but not block the current issue.